### PR TITLE
feat(primitives): Expose Chain ID to RollupConfig Map

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Overhauls the `superhchain-bindings` to expose statically constructed types using `superchain` directory files.

### CODE REVIEW

File: bindings/rust-bindings/src/chain_list.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    pub chain: String,
+}
+
+impl SuperchainParent {
+    /// Returns the chain id for the parent.
+    pub fn chain_id(&self) -> u64 {
+        match self.chain.as_ref() {
+            "mainnet" => 1,
+            "sepolia" => 11155111,
+            "sepolia-dev-0" => 11155420,
```
Comment (clabby) [inline bindings/rust-bindings/src/chain_list.rs:47]: This is not the devnet; `11155420` is OP Sepolia (prod testnet)

File: bindings/rust-bindings/src/lib.rs [hunk 2]
```diff
@@ -6,132 +6,53 @@
...
-        };
-
-        assert_eq!(base.addresses, Some(expected));
-    }
+    // #[test]
+    // fn test_rollup_configs() {
+    //     let op = OP_MAINNET_CONFIG;
+    //     let derived = OPCHAINS.get(&10).unwrap();
+    //     assert_eq!(op.genesis, derived.genesis);
+    // }
```
Comment (clabby) [inline bindings/rust-bindings/src/lib.rs:57]: What's going on here?

File: bindings/rust-primitives/src/rollup_config.rs [hunk 3]
```diff
@@ -32,7 +57,7 @@ pub const OP_SEPOLIA_EIP1559_DEFAULT_ELASTICITY_MULTIPLIER: u128 = 6;
 
 /// Base fee max change denominator for Base Sepolia as defined in the Optimism
 /// [transaction costs](https://community.optimism.io/docs/developers/build/differences/#transaction-costs) doc.
-pub const BASE_SEPOLIA_EIP1559_DEFAULT_ELASTICITY_MULTIPLIER: u128 = 10;
+pub const BASE_SEPOLIA_EIP1559_DEFAULT_ELASTICITY_MULTIPLIER: u128 = 6;
```
Comment (clabby) [inline bindings/rust-primitives/src/rollup_config.rs:60]: This is not right. Base sepolia has a different elasticity multiplier at 10; I fixed this a few weeks back. See https://github.com/paradigmxyz/reth/blob/8c8702be14c6811a4166c279ca1f8df92308044b/crates/primitives-traits/src/constants/mod.rs#L89-L91 + https://github.com/ethereum-optimism/superchain-registry/pull/311

**Review by clabby — CHANGES_REQUESTED** [review]
Nice cleanup overall; The base sepolia elasticity parameter change has to be reverted

File: bindings/rust-primitives/src/rollup_config.rs (hunk 3 above)
Comment (refcell) [reply bindings/rust-primitives/src/rollup_config.rs:60]: Strange, the scr reference config says 6 - will dig in

File: bindings/rust-bindings/src/lib.rs (hunk 2 above)
Comment (refcell) [reply bindings/rust-bindings/src/lib.rs:57]: Ooops rebase error

File: bindings/rust-bindings/src/chain_list.rs (hunk 1 above)
Comment (refcell) [reply bindings/rust-bindings/src/chain_list.rs:47]: Good callout - took an AI to the knee

**Comment by refcell** [comment]
Would like to rebase this on #377 so we don't need the hardcoded list of chain config files, and can instead just read `configs.toml`.

https://github.com/ethereum-optimism/superchain-registry/pull/278/files#diff-717c28977673769ae330a71da195d7584d4d687d972c7c29d4ea6966c33c4c2bR22-R42

**Review by clabby — APPROVED** [review]

**Pull request merged**